lib/matplotlib/colorbar.py

```diff
@@ -873,10 +946,34 @@ def set_alpha(self, alpha):
         self.alpha = alpha
 
     def remove(self):
-        """Remove this colorbar from the figure."""
+        """
+        Remove this colorbar from the figure.
+
+        If the colorbar was created with ``use_gridspec=True`` the previous
+        gridspec is restored.
+        """
         self.ax.inner_ax.remove()
         self.ax.outer_ax.remove()
 
+        self.mappable.callbacksSM.disconnect(self.mappable.colorbar_cid)
+        self.mappable.colorbar = None
+        self.mappable.colorbar_cid = None
+
+        try:
+            ax = self.mappable.axes
+        except AttributeError:
```

Comment:
I guess you aren't testing removing an axes with no mappable?

Reply:
I guess you meant a mappable with no axes? (there's always a mappable, once this PR goes in)  Sure, that seems easy enough to add.